raphaelt-nvidia/sonic-snmpagent
# Vlan snmp

**- What I did**

Restored snmp vlan support per RFC1213, which had been reverted due to inconsistency with RFC2863, and added the missing support for RFC2863.  Added unit tests for RFC2863.

**- How I did it**

Reverted https://github.com/Azure/sonic-snmpagent/pull/191, which was itself a revert of https://github.com/Azure/sonic-snmpagent/pull/169.  Then added code to support vlan in rfc2863.py and unit tests.

Since a long time has elapsed since the original commit and other code has been pushed in the meantime, great care is needed in merging.  Note in particular that mibs.init_sync_d_lag_tables now returns five parameters, the last two of which were added: lag_sai_map and sai_lap_map.  This PR needs one of those maps, and a concurrent commit supporting RFC4363 needs the other, so all the callers were updated and use the one they need.

**- How to verify it**

Unit tests are run via make target/python-wheels/asyncsnmp-2.1.0-py3-none-any.whl.
See also https://github.com/Azure/sonic-snmpagent/pull/169.

**- Description for the changelog**

Restored support for aggregated router interface counters and L3 VLAN counters to RFC1213 MIB, and extended to RFC2863.
